### PR TITLE
Unmaker Buff 

### PR DESCRIPTION
**Pull request by ApplePlastic**
## About The Pull Request
<details>
<summary>
Balance: Gives the GP unmaker some drawbacks; bulk and fire delay, instead of it being a flawless weapon mod. 

Unmakers can deal all forms of damage with no drawbacks included, other than just taking up a mechanical slot. This PR balances it for both energy weapons using Unmakers and ballistics as well, making the weapons a tad larger with also a higher firerate cooldown. 
</summary>
<hr>

	
<hr>
</details>

## Changelog
:cl:
Balance: Gives the GP unmaker some drawbacks; bulk, charge cost and fire delay, instead of it being a flawless weapon mod. Also fixes a typo. 
/:cl:

### CODE REVIEW

**Comment by cdb-is-not-good** [comment]
should probably say what those drawbacks are, if nothing else - for the changelog.

**Comment by ApplePlastic** [comment]
changed
I also wanted to add more energy consumption for energy weapons. I'll wait for some feedback to come through first.

**Comment by ApplePlastic** [comment]
actually, i'll just drop it in now

**Comment by Dimasw99** [comment]
Depending on your Cog. Have the unmaker disintegrate your hand if you fail the check. It is like tripping on under plating but instead your hand is not feeling so good. It doing burn damage is also just fine

**Comment by hex37** [comment]
Not super onboard for this.

Also I'm seeing an unmentioned charge cost nerf?

So, contrary to popular belief as a result of mech laws... The core gameplay loop of R&D is to make _good_ things from research. Getting an unmaker is the most expensive research target R&D has, and also is a bitch to print material wise.

Those are what balance it. Same as hydrogen armor. Same as a bunch of guild gear.

**Comment by AppleGlasstic** [comment]
> This is exactly the opposite of what a buff is. This is called a nerf.

Yes. That’s.. That’s the joke.

**Comment by AppleGlasstic** [comment]
> Not super onboard for this.
> 
> Also I'm seeing an unmentioned charge cost nerf?
> 
> So, contrary to popular belief as a result of mech laws... The core gameplay loop of R&D is to make _good_ things from research. Getting an unmaker is the most expensive research target R&D has, and also is a bitch to print material wise.
> 
> Those are what balance it. Same as hydrogen armor. Same as a bunch of guild gear.


Understandable.
The lack of charge cost makes any energy weapon overpowered with the assistance of a Nano-AI, or anything else that reduces charge cost per fire. I was honestly just going to bar them from energy weapons at first, since you’d think the Unmaker would work the same way as.. any other mod that mentions coating a projectile? Even the Widow does that.

The material cost is mediocre. You can find diamond blades to toss in the lathe pretty much everywhere. 

I’ll edit some of the values when I’m home.

**Comment by ApplePlastic** [comment]
Nevermind. GitHub mobile decided to not be shit.

File: code/modules/projectiles/guns/mods/mods.dm [hunk 1]
```diff
@@ -571,13 +571,15 @@
...
+		UPGRADE_BULK = 2,
 		GUN_UPGRADE_DAMAGE_BRUTE = 10,
 		GUN_UPGRADE_DAMAGE_BURN = 10,
 		GUN_UPGRADE_DAMAGE_TOX = 5,
 		GUN_UPGRADE_DAMAGE_OXY = 5,
 		GUN_UPGRADE_DAMAGE_CLONE = 5,
 		GUN_UPGRADE_DAMAGE_HALLOSS = 5,
-		UPGRADE_BULK = 1
+		GUN_UPGRADE_FIRE_DELAY_MULT = 1.2,
+		GUN_UPGRADE_CHARGECOST = 1.5
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/mods/mods.dm:582]: what if you put it on a bullet gun? Wouldnt that error?

Comment (ApplePlastic) [reply code/modules/projectiles/guns/mods/mods.dm:582]: Eh.. No? NanoAI consumes more power with used on an energy weapon. It should ignore it.

I’ll test it to make sure.

Comment (Trilbyspaceclone) [reply code/modules/projectiles/guns/mods/mods.dm:582]: Good point!

**Review by Trilbyspaceclone — APPROVED** [review]
I like the idea and all that. Tho still needs Hex's okie on this

**Comment by cdb-is-not-good** [comment]
good change, though IMO a bulk change of 2 is pretty negligible. I believe that's- what? Two tiny items worth of extra space?

**Comment by ApplePlastic** [comment]
Not sure if you can add decimals to the bulk change value, so I’ll just leave it out.

**Comment by ApplePlastic** [comment]
> good change, though IMO a bulk change of 2 is pretty negligible. I believe that's- what? Two tiny items worth of extra space?

Edited.
Not sure if you can add decimals to the bulk change value, so I’ll just leave it out and return it back to one.

**Review by hex37 — APPROVED** [review]

**Comment by Stutternov** [comment]
you are doing gods work with this ‘reverse buff’. inshallah, brother.

**Comment by cdb-is-not-good** [comment]
> Not super onboard for this.
> 
> Also I'm seeing an unmentioned charge cost nerf?
> 
> So, contrary to popular belief as a result of mech laws... The core gameplay loop of R&D is to make _good_ things from research. Getting an unmaker is the most expensive research target R&D has, and also is a bitch to print material wise.
> 
> Those are what balance it. Same as hydrogen armor. Same as a bunch of guild gear.

my counter to this is that materials barely help for balance. the state of mining is ridiculously feast/famine. If you have even a single miner there are more materials than we could possibly need and none of them in small amounts. If we don't then it's scraps. That's why i forced r4 to tie the armor to a more singular resource(the core)

**Pull request merged**